src/net_processing.cpp

```diff
@@ -2889,6 +2889,18 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             nCMPCTBLOCKVersion = 1;
             m_connman.PushMessage(&pfrom, msgMaker.Make(NetMsgType::SENDCMPCT, fAnnounceUsingCMPCTBLOCK, nCMPCTBLOCKVersion));
         }
+
+        if (m_txreconciliation) {
+            if (!peer->m_wtxid_relay) {
```

Comment:
Commit "p2p: clear reconciliation state for no-wtxid peers"
Could this be more general so that we also check for `!IsPeerRegistered()` here, forgetting about any pre-registered peers that chose not to send us a SENDRECON before verack - this will be a very frequent case for peers that haven't implemented Erlay or aren't interested in it.